### ISSUE TITLE
build.py --train-modalities is OOM-killed: image_train_resolution defaults to 512 on every preset

### ISSUE THREAD

**Issue by mantreshkhurana** [Spyxpo/framerai#121]
`--train-modalities` is documented as the way to train the image and audio generators on caption pairs, and both GUIDE.md and data/README.md point at the tiny preset:

```bash
python build.py --mode all --size tiny --data-dir data/examples --train-modalities
```

Text training finishes, then the image generator step gets OOM-killed. The cause is `image_train_resolution`, which defaults to 512 in `model/configs/model_config.py` and is not overridden by any preset — including tiny.

Measured on CPU, one forward + backward through the image generator at batch size 1 on a 68 GB machine:

| resolution | peak RSS |
|---|---|
| 64 | 3.0 GB |
| 128 | 37 GB |
| 256 | killed |
| 512 (default) | killed |

Memory grows roughly with the square of the resolution, so 512 is far out of reach on any normal machine, and `build.py` uses `batch_size // 2` (4 on tiny), not 1.

Reproduces with random tensors, so it has nothing to do with the data:

```python
from model.configs import build_preset_config
from model.framer import FramerModel
import torch

cfg = build_preset_config("framer-tiny")
model = FramerModel(cfg); model.train()
out = model(input_ids=torch.randint(0, 100, (1, 64)),
            target_images=torch.randn(1, 3, cfg.image_train_resolution, cfg.image_train_resolution))
out["image_loss"].backward()
```

Worth deciding whether the fix is a per-preset `image_train_resolution` (tiny at 64 or 128), gradient checkpointing on the generator, or both. Whatever we pick, the documented tiny command should actually finish — right now the example datasets added in #120 load fine but the run they're documented for can't complete.

Noticed while reviewing #120.

**Comment by kunaldeo001** [comment]
I’d like to work on Issue #121 — fixing the OOM issue with --train-modalities. Could you please assign it to me?

**Comment by mantreshkhurana** [comment]
Fixed in #123. framer-tiny and framer-tiny-moe set image_train_resolution=64, so the documented command (`python build.py --mode all --size tiny --data-dir data/examples --train-modalities`) completes instead of being OOM-killed at the image stage.

The wider default is unchanged: model_config.py still defaults image_train_resolution to 512, and framer-small, framer-medium, framer-large, framer-3b and framer-8b inherit it, so the same OOM applies to those presets on the modality training path. Tracked separately.
